src/Sarif.Driver/Sdk/SarifLogger.cs

```diff
@@ -32,36 +33,102 @@ public static ToolInfo CreateDefaultToolInfo(string prereleaseInfo = null)
             return toolInfo;
         }
 
+        private static RunInfo CreateRunInfo(
+            IEnumerable<string> analysisTargets,
+            bool computeTargetsHash,
+            IEnumerable<string> invocationInfoTokensToRedact)
+        {
+            var runInfo = new RunInfo();
+
+            if (analysisTargets != null)
+            {
+                runInfo.AnalysisTargets = new List<FileReference>();
+
+                foreach (string target in analysisTargets)
+                {
+                    var fileReference = new FileReference()
+                    {
+                        Uri = target.CreateUriForJsonSerialization(),
+                    };
+
+                    if (computeTargetsHash)
+                    {
+                        string sha256Hash = HashUtilities.ComputeSha256Hash(target) ?? "[could not compute file hash]";
+                        fileReference.Hashes = new List<Hash>(new Hash[]
+                        {
+                            new Hash()
+                            {
+                                Value = sha256Hash,
+                                Algorithm = AlgorithmKind.Sha256,
+                            }
+                        });
+                    }
+                    runInfo.AnalysisTargets.Add(fileReference);
+                }
+            }
+
+            string invocationInfo = Environment.CommandLine;
+
+            if (invocationInfoTokensToRedact != null)
+            {
+                foreach (string tokenToRedact in invocationInfoTokensToRedact)
+                {
+                    invocationInfo = invocationInfo.Replace(tokenToRedact, SarifConstants.RemovedMarker);
+                }
+            }
+            runInfo.InvocationInfo = invocationInfo;
+            runInfo.RunStartTime = DateTime.UtcNow;
+            return runInfo;
+        }
+
+        public SarifLogger(string outputFilePath, bool verbose, ToolInfo toolInfo, RunInfo runInfo)
+            : this (new StreamWriter(new FileStream(outputFilePath, FileMode.Create, FileAccess.Write, FileShare.None)),
+                  verbose,
+                  toolInfo, 
+                  runInfo)
+        {
+
+        }
+
+        public SarifLogger(TextWriter writer, bool verbose, ToolInfo toolInfo, RunInfo runInfo)
+        {
+            Verbose = verbose;
+
+            _textWriter = writer;
+            _jsonTextWriter = new JsonTextWriter(_textWriter);
+
+            // for debugging it is nice to have the following line added.
+            _jsonTextWriter.Formatting = Newtonsoft.Json.Formatting.Indented;
+            _runInfo = runInfo;
+            _issueLogJsonWriter.WriteToolInfo(toolInfo);
```

Comment:
It looks like it's created in the _other_ overload of the ctor (see below, Line 144), but not in this one. This should be DRYd out -- the ctor with the fewer arguments should call the one with the more arguments, or they should both call a common `Init` method.